qualtran/cirq_interop/_cirq_to_bloq.py

```diff
@@ -275,17 +275,15 @@ def _ensure_in_reg_exists(
     for qreg, soq in new_qreg_to_qvar.items():
         if len(in_reg_qubits) > 1 and qreg.qubits and qreg.qubits[0] in in_reg_qubits:
             assert len(qreg.qubits) == 1, "Individual qubits should have been split by now."
-            # Need to split single-qubit Fxp registers which are ambiguous,
-            # these will be appropriately joined later.
-            if isinstance(qreg.dtype, QFxp):
-                soqs_to_join[qreg.qubits[0]] = bb.split(soq=soq)[0]
+            # Cast single bit registers to QBit to preserve signature of later join.
+            if not isinstance(qreg.dtype, QBit):
+                soqs_to_join[qreg.qubits[0]] = bb.add(Cast(qreg.dtype, QBit()), reg=soq)
             else:
                 soqs_to_join[qreg.qubits[0]] = soq
         elif len(in_reg_qubits) == 1 and qreg.qubits and qreg.qubits[0] in in_reg_qubits:
-            # Need to split single-qubit Fxp registers which are ambiguous,
-            # these will be appropriately joined later.
-            if isinstance(qreg.dtype, QFxp):
-                soqs_to_join[qreg.qubits[0]] = bb.split(soq=soq)[0]
+            # Cast single QBit registers to the appropriate single-bit register dtype.
+            if not isinstance(in_reg.dtype, QBit):
+                qreg_to_qvar[in_reg] = bb.add(Cast(QBit(), in_reg.dtype), reg=soq)
```

Comment:
Do we need an assertion here? Is guaranteed that `soq` will always be of type `QBit` ?

Reply:
I added an assertion.